packages/iframe-execution-environment-service/src/IframeExecutionService.ts

```diff
@@ -129,20 +124,12 @@ export class IframeExecutionService extends AbstractExecutionService<EnvMetadata
     };
   }
 
-  private _createWindow(
-    uri: string,
-    jobId: string,
-    timeout: number,
-  ): Promise<Window> {
+  private _createWindow(uri: string, jobId: string): Promise<Window> {
     const iframe = document.createElement('iframe');
-    return new Promise((resolve, reject) => {
-      const errorTimeout = setTimeout(() => {
-        iframe.remove();
-        reject(new Error(`Timed out creating iframe window: "${uri}"`));
-      }, timeout);
+    // This may run forever if the iframe never loads, but we wrap this call in a timeout elsewhere
```

Comment:
This could perhaps go in a docstring for the function instead. We should definitely add a docstring to the `_initJob` and `executeSnap` methods on the base class to indicate that `_initJob` may never resolve.

Finally, which timeout are you referring to here?

Reply:
Added doc strings and clarified that the timeout is added in SnapController when waiting for initialization.